scripts/cfFindIP.sh

```diff
@@ -27,21 +42,52 @@ scriptDir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 resultDir="$scriptDir/../result"
 resultFile="$resultDir/$now-result.cf"
 
+#check if expected output folder exists and create if it's not availbe
+if [ ! -d "$resultDir" ]; then
+    mkdir -p "$resultDir"
+fi
+
+
 # Function fncCheckSubnet
 # Check Subnet
 function fncCheckSubnet {
-	local ipList resultFile 
+	local ipList resultFile timeoutCommand
 	ipList="$1"
 	resultFile="$2"
+# Check if 'timeout' command is available
+if command -v timeout >/dev/null 2>&1; then
+    timeoutCommand="timeout"
+else
+    # Check if 'gtimeout' command is available
+    if command -v gtimeout >/dev/null 2>&1; then
+        timeoutCommand="gtimeout"
+    else
+        # Check if 'perl-timeout' command is available
+        if command -v perl-timeout >/dev/null 2>&1; then
+            timeoutCommand="perl-timeout"
+        else
+            # Check if 'python-timeout' command is available
+            if command -v python-timeout >/dev/null 2>&1; then
+                timeoutCommand="python-timeout"
+            else
+                echo >&2 "I require 'timeout' command but it's not installed. Please install 'timeout' or an alternative command like 'gtimeout' or 'perl-timeout' or 'python-timeout' and try again."
+                exit 1
+            fi
+        fi
+    fi
+fi
+
```

Comment:
This block needs to be adjusted with space and tabs to be more human readable

Reply:
If you check timeout command upper then you need to just set here no more check

Reply:
And also making the code compatible with non-basic tools like perl-timeout and python-timeout is not good idea
timeout for Linux and gtimout for Mac is enough

Reply:
done